docs/madr/decisions/018-fault-injection-policy.md

```diff
@@ -0,0 +1,258 @@
+
+```yaml
+spec:
+  sources:
+    - match:
+        kuma.io/service: frontend_default_svc_80
+        kuma.io/protocol: http
+  destinations:
+    - match:
+        kuma.io/service: backend_default_svc_80
+        kuma.io/protocol: http
+  conf:
+    abort:
+      httpStatus: 500
+      percentage: 50
+    delay:
+      percentage: 50.5
+      value: 5s
+    responseBandwidth:
+      limit: 50 mbps
+      percentage: 50
+```
+
+The configuration translates to the Envoy configuration in Http's filters:
+
+```yaml
+{
+"name": "envoy.filters.http.fault",
+"typed_config": {
+  "@type": "type.googleapis.com/envoy.extensions.filters.http.fault.v3.HTTPFault",
+  "delay": {
+  "fixed_delay": "5s",
+  "percentage": {
+    "numerator": 505000,
+    "denominator": "TEN_THOUSAND"
+  }
+  },
+  "abort": {
+  "http_status": 500,
+  "percentage": {
+    "numerator": 50
+  }
+  },
+  "headers": [
+  {
+    "name": "x-kuma-tags",
+    "safe_regex_match": {
+    "google_re2": {},
+    "regex": ".*&kuma.io/protocol=[^&]*http[,&].*&kuma.io/service=[^&]*frontend_default_svc_80[,&].*"
+    }
+  }
+  ],
+  "response_rate_limit": {
+  "fixed_limit": {
+    "limit_kbps": "50000"
+  },
+  "percentage": {
+    "numerator": 50
+  }
+  }
+}
+```
+
+### Specification
+
+Fault injection supports only `HTTP` protocol and is configured in `HttpFilter`. We can use the header `x-kuma-tags` to match the specific source. In the case of `MeshHealthCheck` which doesn't set `x-kuma-tags` we can put `x-kuma-tags` or create `MeshFaultInjection` that allows only `Mesh` in `From` section.
+
+#### Top level
+
+Top-level targetRef can have all available kinds:
+```yaml
+targetRef:
+ kind: Mesh|MeshSubset|MeshService|MeshServiceSubset|MeshGatewayRoute
+ name: ...
+```
+Fault injection can be configured on inbound and outbound.
+
+#### From level
+
+We can allow configuration from different sources.
+```yaml
+from:
+ - targetRef:
+     kind: Mesh|MeshSubset|MeshService|MeshServiceSubset
+     name: ...
+```
+The only issue is `MeshHealthCheck` which doesn't set the `x-kuma-tags` flag and might not match the specific fault injection configuration. We have 2 options:
+
+* set `x-kuma-tags` header by default for `MeshHealthCheck`
+* change the scope of `From` to `Mesh` so the policy will affect all sources
+
+#### To level
+
+We can allow configuration to different destinations.
+```yaml
+from:
+ - targetRef:
+     kind: Mesh|MeshSubset|MeshService|MeshServiceSubset
+     name: ...
+```
+
+#### Configuration
+
+```yaml
+default:
+  http:
+    abort:
+      httpStatus: 500
+      percentage: "50" # K8s shows warning when using float so we agree to use string and parse in the code
+    delay:
+      percentage: "50.5" # K8s shows warning when using float so we agree to use string and parse in the code
+      value: 5s
+    responseBandwidth:
+      limit: 50 mbps
+      percentage: "50" # K8s shows warning when using float so we agree to use string and parse in the code
```

Comment:
Where did this conversation happen? Do we have a fix for MeshCircuitBreaker and MeshTrafficTrace and the memo already?

Reply:
We don't have fixes yet, https://github.com/kumahq/kuma/pull/5608#discussion_r1067843471 was a suggestion, and I haven't seen more replies/objections.